shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -341,23 +341,33 @@ - (void)updateViewportMetrics {
   });
 }
 
+- (bool)isWindowFullscreen {
+  UIWindow *window = self.view.window;
+  return CGRectEqualToRect(window.frame, window.screen.bounds);
+}
+
+- (CGFloat)statusBarPadding {
+  // If not fullscreen, assume we don't want padding.
+  if (![self isWindowFullscreen]) {
+    return 0.0;
+  }
+
+  UIScreen *screen = self.view.window.screen;
+  CGRect statusFrame = [UIApplication sharedApplication].statusBarFrame;
+  CGRect viewFrame = [self.view convertRect:self.view.bounds
+                          toCoordinateSpace:screen.coordinateSpace];
+  CGFloat padding = statusFrame.origin.y + statusFrame.size.height - viewFrame.origin.y;
+  return MAX(padding, 0.0);
```

Comment:
Could probably avoid the `MAX()` by doing a `CGRectIntersectsRect()` check, though this seems cheaper.